protocol/messenger_messages.go

```diff
@@ -82,6 +82,71 @@ func (m *Messenger) EditMessage(ctx context.Context, request *requests.EditMessa
 	return response, nil
 }
 
+func (m *Messenger) DeleteMessageAndSend(ctx context.Context, messageID string) (*MessengerResponse, error) {
+	message, err := m.persistence.MessageByID(messageID)
+	if err != nil {
+		return nil, err
+	}
+
+	if message.From != common.PubkeyToHex(&m.identity.PublicKey) {
+		return nil, ErrInvalidEditOrDeleteAuthor
+	}
+
```

Comment:
Should the user be allowed to delete this kind of messages? (dunno if it's possible to send a message of this type)
```
	0:  "UNKNOWN_CONTENT_TYPE",
	3:  "STATUS",
	5:  "TRANSACTION_COMMAND",
	6:  "SYSTEM_MESSAGE_CONTENT_PRIVATE_GROUP",
	10: "SYSTEM_MESSAGE_GAP",
```

Reply:
No, none of them, I am not sure about `STATUS`, I don't think it's used?

Reply:
Nice catch.

I used a whitelist approach, that way, if we ever add a new message type that cannot be deleted, it will be denied. I feel like it's better to block deleting more than letting through a delete that would break the app (so stability over usability in case of new types)